### PR TITLE
Add initial implementation

### PR DESCRIPTION
**Pull request by ykulazhenkov**

### CODE REVIEW

File: cmd/network-operator-init-container/app/app.go [hunk 1]
```diff
@@ -76,9 +90,126 @@ func NewNetworkOperatorInitContainerCommand() *cobra.Command {
...
+
+// NodeReconciler reconciles Node object
+type NodeReconciler struct {
+	ErrCh              chan error
+	SafeLoadAnnotation string
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+// Reconcile contains logic to sync IPPool objects
```
Comment (rollandf) [inline cmd/network-operator-init-container/app/app.go:176]: sync Node objects

**Comment by ykulazhenkov** [comment]
want to apply some changes to the code, move PR to Draft

File: cmd/network-operator-init-container/app/app.go (hunk 1 above)
Comment (ykulazhenkov) [reply cmd/network-operator-init-container/app/app.go:176]: thx, I will fix this

Comment (ykulazhenkov) [reply cmd/network-operator-init-container/app/app.go]: fixed

**Comment by ykulazhenkov** [comment]
I update the code to use config file instead of command line arguments

File: cmd/network-operator-init-container/app/app.go [hunk 2]
```diff
@@ -76,9 +89,139 @@ func NewNetworkOperatorInitContainerCommand() *cobra.Command {
...
+		reqLog.Info("annotation removed, unblock loading")
+		writeCH(r.ErrCh, nil)
+		return ctrl.Result{}, nil
+	}
+	reqLog.Info("annotation still present, waiting")
+
+	return ctrl.Result{RequeueAfter: time.Second * 5}, nil
+}
+
+func writeCH(ch chan error, err error) {
```
Comment (e0ne) [inline cmd/network-operator-init-container/app/app.go:215]: nit: writeCH -> writeCh

File: pkg/config/config.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+	"os"
+
+	"k8s.io/apimachinery/pkg/util/json"
+)
+
+// FromFile reads configuration from the file
+func FromFile(path string) (Config, error) {
+	data, err := os.ReadFile(path)
+	if err != nil {
+		return Config{}, fmt.Errorf("failed to read data from the provided file: %v", err)
```
Comment (e0ne) [inline pkg/config/config.go:27]: IMO, we shouldn't return an empty config. it's enouph to return `nil, fmt.Errorf("failed to read data from the provided file: %v", err)`

File: pkg/config/config.go [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+	}
+	cfg := Config{}
+	if err := json.Unmarshal(data, &cfg); err != nil {
+		return Config{}, fmt.Errorf("failed to unmarshal configuration: %v", err)
+	}
+	if err := cfg.Validate(); err != nil {
+		return Config{}, fmt.Errorf("configuration is invalid: %v", err)
```
Comment (e0ne) [inline pkg/config/config.go:34]: the same as above

File: pkg/config/config.go [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+// FromFile reads configuration from the file
+func FromFile(path string) (Config, error) {
+	data, err := os.ReadFile(path)
+	if err != nil {
+		return Config{}, fmt.Errorf("failed to read data from the provided file: %v", err)
+	}
+	cfg := Config{}
+	if err := json.Unmarshal(data, &cfg); err != nil {
+		return Config{}, fmt.Errorf("failed to unmarshal configuration: %v", err)
```
Comment (e0ne) [inline pkg/config/config.go:31]: the same as above

**Review by e0ne — APPROVED** [review]

**Review by rollandf — APPROVED** [review]
LGTM

**Review by rollandf — APPROVED** [review]

File: cmd/network-operator-init-container/app/app.go (hunk 2 above)
Comment (ykulazhenkov) [reply cmd/network-operator-init-container/app/app.go]: fixed

File: pkg/config/config.go (hunk 4 above)
Comment (ykulazhenkov) [reply pkg/config/config.go]: ok, let's return the pointer. fixed

File: pkg/config/config.go (hunk 5 above)
Comment (ykulazhenkov) [reply pkg/config/config.go]: fixed

**Pull request merged**